src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -413,6 +414,770 @@ public static class C
+    }
+
+    public int A;
+    public int B;
+}
+
+[My(A=1, B=2)]
+class Program
+{
+    static void Main()
+    {
+        typeof(Program).GetCustomAttribute(typeof(MyAttribute));
+    }
+}";
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: "'<default1>', '<default0>'");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpressionInAttributeConstructor_LangVersion9()
+        {
+            string source = @"
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+class MyAttribute : Attribute
+{
+    public MyAttribute(int a = 1, [CallerArgumentExpression(""a"")] string expr_a = """")
+    {
+        Console.WriteLine($""'{a}', '{expr_a}'"");
+    }
+}
+
+[My(1+2)]
+class Program
+{
+    static void Main()
+    {
+        typeof(Program).GetCustomAttribute(typeof(MyAttribute));
+    }
+}";
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.Regular9);
+            compilation.VerifyDiagnostics(
+                // (14,4): error CS8652: The feature 'caller argument expression' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // [My(1+2)]
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "(1+2)").WithArguments("caller argument expression").WithLocation(14, 4));
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpression_ImplicitConversionFromStringExists()
+        {
+            string source = @"
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+class MyAttribute : Attribute
+{
+    public MyAttribute(int a = 1, [CallerArgumentExpression(""a"")] object expr_a = null)
+    {
+        Console.WriteLine($""'{a}', '{expr_a}'"");
+    }
+}
+
+[My(1+2)]
+class Program
+{
+    static void Main()
+    {
+        typeof(Program).GetCustomAttribute(typeof(MyAttribute));
+    }
+}";
+            var compilation = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseExe, parseOptions: TestOptions.RegularPreview);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: @"'3', '1+2'");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestArgumentExpression_ImplicitConversionFromStringDoesNotExists()
+        {
+            string source = @"
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+class MyAttribute : Attribute
+{
+    public MyAttribute(int a = 1, [CallerArgumentExpression(""a"")] int expr_a = 0)
```

Comment:
>[CallerArgumentExpression(""a"")] [](http://example.com/codeflow?start=34&length=33)

Should we cover a scenario whan an attribute like that is applied in metadata?

Reply:
@AlekseyTs The attribute definition is an error in this case, so how would I emit it as a metadata reference?

Reply:
> The attribute definition is an error in this case, so how would I emit it as a metadata reference?

The simplest way would be to test with IL source. See ```CreateCompilationWithIL``` helper for example.

Reply:
@AlekseyTs Added the test, not sure if the current behavior is what you'd expect or not. Can you please take a look? Thanks!